### PR TITLE
[Fixes] Several small fixes for the Publishing logic

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Added missing 'ErrorAction'
- Fixed incorrect array count validation
- Aligned naming

Successfully validated that it doesn't fail on missing ACR & correctly creates a new one once necessary
image

| Pipeline |
| - |
| [AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1 [hunk 1]
```diff
@@ -71,7 +71,7 @@ function Get-ModulesMissingFromPrivateBicepRegistry {
         # Get all children
         $availableModuleTemplatePaths = (Get-ChildItem -Path (Split-Path $TemplateFilePath) -Recurse -Include @('deploy.bicep', 'deploy.json')).FullName
 
-        if (-not (Get-AzContainerRegistry -Name $BicepRegistryName -ResourceGroupName $BicepRegistryRgName)) {
+        if (-not (Get-AzContainerRegistry -Name $BicepRegistryName -ResourceGroupName $BicepRegistryRgName -ErrorAction 'SilentlyContinue')) {
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1:74]: Does the fix allow the creation of the rg as well if not existing?

Comment (AlexanderSehr) [reply utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1:74]: The `Get-Modules(...)` functions don't create anything. They just check if the resouce is existing at all - and if not consider all modules worth publishing. It's only the following `Publish` functions that create the resources.

Comment (eriqua) [reply utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1:74]: I meant the general fix, I'm not referring to this file or line in particular. That's just to create a comment that can be easily resolved.

I understand this PR validates that a missing ACR does not cause the publishing to fail. My question is if the same stands for the RG hosting the specified ACR. So if neither the ACR nor the ACR RG exist, does the publishing create both?

Comment (AlexanderSehr) [reply utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1:74]: Yes it does. In the corresponding files for the registry and templates specs, the resources are fetched and, if not existing, created

Comment (AlexanderSehr) [reply utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1:74]: Woop woop

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**